.github/libs/GitUtils.js

```diff
@@ -48,26 +48,39 @@ function getMergeLogsAsJSON(fromRef, toRef) {
 /**
  * Parse merged PRs, excluding those from irrelevant branches.
  *
- * @param {Array<String>} commitMessages
+ * @param {Array<Object<{commit: String, subject: String, authorName: String}>>} commits
  * @returns {Array<String>}
  */
-function getValidMergedPRs(commitMessages) {
-    return _.reduce(
-        commitMessages,
-        (mergedPRs, commitMessage) => {
-            if (!_.isString(commitMessage)) {
-                return mergedPRs;
-            }
+function getValidMergedPRs(commits) {
+    const mergedPRs = new Set();
+    _.each(commits, (commit) => {
+        const author = commit.authorName;
+        if (author === 'OSBotify') {
+            return;
+        }
 
-            const match = commitMessage.match(/Merge pull request #(\d+) from (?!Expensify\/(?:main|version-|update-staging-from-main|update-production-from-staging))/);
-            if (!_.isNull(match) && match[1]) {
-                mergedPRs.push(match[1]);
-            }
+        const message = commit.subject;
+        if (!_.isString(message)) {
```

Comment:
This safety check might not be necessary – the subject should always be a non-empty string

Reply:
Ah commented on the actions file, was asking about this, it would be great to add a comment about this just to make it easier to read for anyone.

Reply:
Removed this safety check, hopefully shouldn't be necessary as long as every commit has a subject, which I think is a safe assumption.